pyproject.toml

```diff
@@ -106,8 +106,10 @@ uvicorn = { extras = ["standard"], version = ">=0.22.0", optional = true }
 aiosqlite = "*"
 asyncmy = "*"
 asyncpg = "*"
+beanie = ">=1.21.0"
```

Comment:
not sure I understand how this is related with anyio
same for the changes [tests/unit/test_contrib/test_pydantic/test_beanie_integration.py](https://github.com/litestar-org/litestar/pull/2287/files#diff-fd74b705880373004fc0f91b302e044d7496cc47477c19be3c548e4244f7a62f)

Reply:
this was also failing in the pipeline, so I installed it. apparently it now supports pydantic v2, so no longer a need to drop it out.